test/binarta-angular.spec.js

```diff
@@ -2043,6 +2043,19 @@
                     $ctrl.$onInit();
                     expect($ctrl.onSelect).toHaveBeenCalledWith('payment-method');
                 });
+
+                it('should check if the paymentMethodIsAvailable', function () {
+                  expect($ctrl.isPaymentMethodAvailable('payment-method')).toBe(true);
+                  expect($ctrl.isPaymentMethodAvailable('no-payment-method-available')).toBe(false);
+                });
+
+                it('should set Paypal as default when this payment method is available', function () {
```

Comment:
Hier hebben we dus het hart van de zaak.

Mijn voornaamste bezwaar dat ik kan maken is dat hier een beslissing tot het selecteren van een default gemaakt wordt en dat het maken van die beslissing op de verkeerde plaats gebeurd. Of toch op zijn minst op de verkeerde manier.

Wat bedoel ik daarmee.
Angular.

Angular is een framework waar we geen controle over hebben en zelfs van weg willen op lange termijn. Denk Angular 2, 3, 4,...
Dus we willen zo min mogelijk beslissingen koppelen aan code die we in zo een situatie moeten weg gooien.

Wat ik dus zou doen is in binarta.js de nodige aanpassingen maken zodat het eenvoudig is om te zien welke betalings methode de default keuze moet zijn.

Met andere woorden hier:
`binarta.application.profile().availablePaymentMethods`

Hier heb je denk ik wat opties.

1. het resultaat kan iets zijn in de volgende stijl `[{method:"paypal", "default":true}, {method:"wire-transfer"}]`.
1. of misschien is er een functie waarmee je de default kan opvragen en dus de beschikbare methodes mee testen. `binarta.application.profile().getDefaultPaymentMethod()`

Op deze manier wordt de code in Angular beperkt tot het opzetten van de UI widgets volgens de instructies van de echte code. En ik zou denken dat we zelfs het bepalen van de default dan naar de server kunnen verplaatsen zodat we het maken van die beslissing niet moeten dupliceren in alle UI's.